packages/typescript-estree/src/parseSettings/resolveProjectList.ts

```diff
@@ -56,15 +56,15 @@ export function resolveProjectList(
 
   const projectFolderIgnoreList = (
     options.projectFolderIgnoreList ?? ['**/node_modules/**']
-  )
-    .reduce<string[]>((acc, folder) => {
-      if (typeof folder === 'string') {
-        acc.push(folder);
-      }
-      return acc;
-    }, [])
-    // prefix with a ! for not match glob
-    .map(folder => (folder.startsWith('!') ? folder : `!${folder}`));
+  ).reduce<string[]>((acc, folder) => {
+    if (typeof folder === 'string') {
+      acc.push(
+        // prefix with a ! for not match glob
+        folder.startsWith('!') ? folder : `!${folder}`,
+      );
+    }
+    return acc;
+  }, []);
```

Comment:
A bonus change, replace the `.reduce.map` chain with a single `.reduce` to avoid unnecessary iteration.